src/solvers/jesse_dg/ESDG_2D.jl

```diff
@@ -0,0 +1,281 @@
+
+function Trixi.create_cache(mesh::UnstructuredMesh, equations, rd::RefElemData, RealT, uEltype)
+
+    @unpack VXYZ,EToV = mesh
+    md = MeshData(VXYZ...,EToV,rd)
+    md = make_periodic(md,rd)
+
+    # for flux differencing on general elements
+    Qrhskew,Qshskew,VhP,Ph = build_hSBP_ops(rd)
+    QrhskewTr = Matrix(Qrhskew')
+    QshskewTr = Matrix(Qshskew')
+
+    # tmp variables for entropy projection
+    Uq = StructArray(staticzip(SVector{4},ntuple(_->similar(md.xq),4)))
+    VUq = similar(Uq)
+    VUh = StructArray(staticzip(SVector{4},ntuple(_->similar([md.xq;md.xf]),4)))
+    Uh = similar(VUh)
+
+    cache = (;md,
+            QrhskewTr,QshskewTr,VhP,Ph,
+            Uq,VUq,VUh,Uh)
+
+    return cache
+end
+
+@inline function tmap!(f,out,x)
+    Trixi.@threaded for i = 1:length(x)
+        @inbounds out[i] = f(x[i])
+    end
+end
+
+project_and_store!(y,x) = mul!(y,Ph,x) # can't use matmul! b/c its applied to a subarray
+
+## workaround for matmul! with threads https://discourse.julialang.org/t/odd-benchmarktools-timings-using-threads-and-octavian/59838/5
+@inline function bmap!(f,out,x)
+    @batch for i = 1:length(x)
+        @inbounds out[i] = f(x[i])
+    end
+end
+
+@inline function compute_entropy_projection!(Q,rd::RefElemData,cache,eqn) 
+    @unpack Vq = rd    
+    @unpack VhP,Ph = cache
+    @unpack Uq, VUq, VUh, Uh = cache
+
+    # entropy projection - should be zero alloc
+    StructArrays.foreachfield((uout,u)->mul!(uout,Vq,u),Uq,Q) # matmul causes issue?
+    tmap!(u->cons2entropy(u,eqn),VUq,Uq) # 77.5μs
+    StructArrays.foreachfield((uout,u)->mul!(uout,VhP,u),VUh,VUq) # matmul slower 
+    tmap!(v->entropy2cons(v,eqn),Uh,VUh) # 327.204 μs
+
+    # StructArrays.foreachfield((uout,u)->matmul!(uout,Vq,u),Uq,Q)  
+    # bmap!(u->cons2entropy(u,eqn),VUq,Uq) # 77.5μs
+    # StructArrays.foreachfield((uout,u)->matmul!(uout,VhP,u),VUh,VUq) 
+    # bmap!(v->entropy2cons(v,eqn),Uh,VUh) # 327.204 μs
+
+    Nh,Nq = size(VhP)
+    Uf = view(Uh,Nq+1:Nh,:) # 24.3 μs
+
+    return Uh,Uf
+end
+
+@inline function max_abs_speed_normal(UL, UR, normal, equations::CompressibleEulerEquations2D) 
+    # Calculate primitive variables and speed of sound
+    ρu_n_L = UL[2]*normal[1] + UL[3]*normal[2]
+    ρu_n_R = UR[2]*normal[1] + UR[3]*normal[2]
+    uL = (UL[1],ρu_n_L,UL[4])
+    uR = (UR[1],ρu_n_R,UR[4])
+    return Trixi.max_abs_speed_naive(uL,uR,nothing,CompressibleEulerEquations1D(1.4))
+end
+
+function Trixi.rhs!(dQ, Q::StructArray, t,
+                    mesh::UnstructuredMesh, equations::CompressibleEulerEquations2D,
+                    initial_condition, boundary_conditions, source_terms,
+                    rd::RefElemData, cache)
+
+    @unpack md = cache
+    @unpack QrhskewTr,QshskewTr,VhP,Ph = cache
+    @unpack rxJ,sxJ,ryJ,syJ,J,nxJ,nyJ,sJ,mapP = md
+    @unpack Vq,wf = rd
+
+    Nh,Nq = size(VhP)
+
+    @inline F(orientation) = (uL,uR)->Trixi.flux_chandrashekar(uL,uR,orientation,CompressibleEulerEquations2D(1.4))
```

Comment:
```suggestion
    @inline F(orientation) = (uL,uR)->Trixi.flux_chandrashekar(uL,uR,orientation,equations)
```
Does that work for you?

Reply:
It works, but it also results in allocations: 
```julia
julia> @btime F($1)($(Uh[1]),$(Uh[2]))
  148.676 ns (4 allocations: 176 bytes)
```
Old approach
```julia
julia> @btime Trixi.flux_chandrashekar($(Uh[1]),$(Uh[1].*2),$1,$(equations))
  23.815 ns (0 allocations: 0 bytes)
```

Reply:
I think this is an issue with non-constant binding of `F` https://julialang.slack.com/archives/CF53T1DU4/p1619751948298700

Reply:
Nvm, this might just be an issue with the way I'm benchmarking (was benchmarking in the REPL, and because REPL variables are in global scope, this impacted timings)